newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -268,17 +270,52 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     }
   }
 
-  _persistValuesToLocalStorage(preferences: Preferences) {
+  _persistValuesToLocalStorage(values: PreferencesValues) {
     try {
-      localStorage.setItem(
-        LocalStorageItem,
-        JSON.stringify(preferences.values)
-      );
+      localStorage.setItem(LocalStorageItem, JSON.stringify(values));
     } catch (e) {
       console.warn('Unable to persist preferences', e);
     }
+  }
+
+  _getLastUsedPath(project: gdProject, kind: ResourceKind) {
+    try {
+      const projectName = project.getName();
+      const values = this.state.values;
+      if (values) {
+        const curProjectPaths = values.lastUsedPath[projectName];
+        if (curProjectPaths) {
+          return curProjectPaths[kind];
+        }
+      }
```

Comment:
You're returning undefined, but you declared elsewhere that your function is returning a `string`. You must make a conscious choice and stick to it to avoid crashes/bugs.